src/ui/wxWidgets/TreeCtrl.cpp

```diff
@@ -901,6 +1013,436 @@ void TreeCtrl::OnKeyDown(wxTreeEvent& evt)
   evt.Skip();
 }
 
+TreeScrollTimer::TreeScrollTimer()
+{
+  m_owner = NULL;
+}
+
+void TreeScrollTimer::Notify()
+{
+  wxASSERT(m_owner);
+  m_owner->CheckScrollList();
+}
+
+TreeCollapseTimer::TreeCollapseTimer()
+{
+    m_owner = NULL;
+}
+
+void TreeCollapseTimer::Notify()
+{
+  wxASSERT(m_owner);
+  m_owner->CheckCollapsEntry();
+}
+
+void TreeCtrl::CheckScrollList()
+{
+  if(m_isDragging) {
+    int width, height;
+    wxPoint mousePt = ScreenToClient(wxGetMousePosition());  // Find where the mouse is in relation to the (exited) pane
+    
+    GetSize(&width, &height);  // Store window dimensions
+    
+    if((mousePt.x < 0) || (mousePt.x >= width)) { // Not in same area
+      resetScrolling();
+      return;
+    }
+    
+    int ThumbVert = GetScrollThumb (wxVERTICAL);
+    int PosVert = GetScrollPos (wxVERTICAL);
+    int RangeVert = GetScrollRange (wxVERTICAL);
+    wxEventType commandType = wxEVT_NULL;
+    
+    if(mousePt.y <= 0) { // We're above the pane
+      StopAutoScrolling();
+      if(PosVert > 0) {
+        commandType = wxEVT_SCROLLWIN_PAGEUP;
+      }
+    }
+    else if(mousePt.y > height) {  // We're below the pane
+      StopAutoScrolling();
+      if((PosVert + ThumbVert) < RangeVert) {
+        commandType = wxEVT_SCROLLWIN_PAGEDOWN;
+      }
+    }
+    else if(mousePt.y <= m_upper_scroll_limit) {
+      if(PosVert > 0) {
+        commandType = wxEVT_SCROLLWIN_LINEUP;
+      }
+    }
+    else if(mousePt.y >= m_lower_scroll_limit) {
+      if((PosVert + ThumbVert) < RangeVert) {
+        commandType = wxEVT_SCROLLWIN_LINEDOWN;
+      }
+    }
+    
+    if(commandType != wxEVT_NULL) {
+      wxScrollWinEvent scrollEvent( commandType, GetId(), 0);
+      GetEventHandler()->ProcessEvent(scrollEvent);
+      m_scroll_timer.Start( TreeScrollTimer::DELAY, true );
+    }
+    else {
+      resetScrolling();
+    }
+  }
+}
+
+void TreeCtrl::CheckCollapsEntry()
+{
+  if(! m_last_mice_item_in_drag_and_drop.IsOk()) return;
+  
+  int width, height;
+  wxPoint mousePt = ScreenToClient(wxGetMousePosition());  // Find where the mouse is in relation to the (exited) pane
+  
+  GetSize(&width, &height);  // Store window dimensions
+  
+  if((mousePt.x < 0) || (mousePt.x >= width) || (mousePt.y < 0) || (mousePt.y >= height)) { // Not in same area
+    m_last_mice_item_in_drag_and_drop = NULL;
```

Comment:
I would suggest using 'nullptr'.

Reply:
It's not 100% used as nullptr throughout the source tree - NULL can be seen at other places also for pointers - but I did as proposed for the new code.

Reply:
It is not so dramatic having NULL in the source code, but there have already been some replacements of NULL by nullptr in the past. However, there could still be some occurrences of NULL here and there in the source code. It would at least be  good if NULL were no longer introduced. - Thanks for the changes.